litestar/contrib/pydantic/pydantic_schema_plugin.py

```diff
@@ -141,6 +141,48 @@ def __init__(self, prefer_alias: bool = False) -> None:
     @staticmethod
     def is_plugin_supported_type(value: Any) -> bool:
         return isinstance(value, _supported_types) or is_class_and_subclass(value, _supported_types)  # type: ignore
+    
+    def get_field_name(self, field_definition: FieldDefinition, schema_creator: SchemaCreator) -> str:
+        model_fields: dict[str, pydantic.fields.FieldInfo] = {
+            k: getattr(f, "field_info", f)
+            for k, f in getattr(field_definition.annotation, "__fields__", getattr(field_definition.annotation, "model_fields", {})).items()
+        }
+
+        annotation_hints = get_type_hints(field_definition.annotation, include_extras=True)
```

Comment:
```suggestion
        annotation_hints = field_definition.get_type_hints(include_extras=True)
```
`FieldDefinition` has a method for exactly this!